### PR TITLE
feat(web-client): add search service

### PR DESCRIPTION
**Pull request by billguo99**
Adding search service to connect to the 2 backend APIs that connects to our CosmosDB. Also add wallet data into db when new wallet is created. Tested on ntls-staging.

### CODE REVIEW

**Comment by Claire-Bear** [comment]
Task linked: [CU-315ghwb Add search frontend code](https://app.clickup.com/t/315ghwb)

**Review by IscoRuta98 — APPROVED** [review]
LGTM!

File: web-client/src/app/state/session.service.ts [hunk 1]
```diff
@@ -58,6 +60,15 @@ export class SessionService {
...
         const wallet = result.Created;
         this.sessionStore.update({ wallet, pin });
+        try {
+          await this.searchService.insertWalletAddress({
+            wallet_id: wallet.wallet_id,
+            phone_number: wallet.phone_number,
+            owner_name: wallet.owner_name,
+          });
+        } catch (err) {
+          console.log(err);
```
Comment (jdvlio) [inline web-client/src/app/state/session.service.ts:70]: I believe this should be `console.error`

**Review by jdvlio — CHANGES_REQUESTED** [review]

File: web-client/src/app/state/session.service.ts (hunk 1 above)
Comment (billguo99) [reply web-client/src/app/state/session.service.ts:70]: No, there are other places in our code that has console.log(err) - https://github.com/ntls-io/nautilus-wallet/blob/23a328a827c853028880e7222da5a5487246cbbf/web-client/src/app/views/register/register.page.ts#L109

**Review by sonasi — APPROVED** [review]

File: web-client/src/app/state/session.service.ts (hunk 1 above)
Comment (jdvlio) [reply web-client/src/app/state/session.service.ts:70]: I see.  I won't pretend to understand it but I assume it relates to how actionable the error message is.

**Review by jdvlio — APPROVED** [review]
Seems good

**Pull request merged**